modules/swagger-codegen/src/main/resources/Java/api.mustache

```diff
@@ -86,12 +86,26 @@ public class {{classname}} {
     final String {{localVariablePrefix}}contentType = {{localVariablePrefix}}apiClient.selectHeaderContentType({{localVariablePrefix}}contentTypes);
 
     String[] {{localVariablePrefix}}authNames = new String[] { {{#authMethods}}"{{name}}"{{#hasMore}}, {{/hasMore}}{{/authMethods}} };
+    {{#responses}}
+    {{#isDefault}}
+
+    {{#isBinary}}
+    byte[] {{localVariablePrefix}}response = null;
+    {{localVariablePrefix}}response = {{localVariablePrefix}}apiClient.invokeBinaryAPI({{localVariablePrefix}}path, "{{httpMethod}}", {{localVariablePrefix}}queryParams,{{localVariablePrefix}} postBody, {{localVariablePrefix}}postBinaryBody, {{localVariablePrefix}}headerParams, {{localVariablePrefix}}formParams, {{localVariablePrefix}}accept, {{localVariablePrefix}}contentType, {{localVariablePrefix}}authNames);
+    return {{localVariablePrefix}}response;
+    {{/isBinary}}
+
+    {{^isBinary}}
     {{#returnType}}
     TypeRef {{localVariablePrefix}}returnType = new TypeRef<{{{returnType}}}>() {};
-    return {{localVariablePrefix}}apiClient.invokeAPI({{localVariablePrefix}}path, "{{httpMethod}}", {{localVariablePrefix}}queryParams, {{localVariablePrefix}}postBody, {{localVariablePrefix}}headerParams, {{localVariablePrefix}}formParams, {{localVariablePrefix}}accept, {{localVariablePrefix}}contentType, {{localVariablePrefix}}authNames, {{localVariablePrefix}}returnType);
+    return {{localVariablePrefix}}apiClient.invokeAPI({{localVariablePrefix}}path, "{{httpMethod}}", {{localVariablePrefix}}queryParams, {{localVariablePrefix}}postBody, {{localVariablePrefix}}postBinaryBody, {{localVariablePrefix}}headerParams, {{localVariablePrefix}}formParams, {{localVariablePrefix}}accept, {{localVariablePrefix}}contentType, {{localVariablePrefix}}authNames, {{localVariablePrefix}}returnType);
     {{/returnType}}{{^returnType}}
-    {{localVariablePrefix}}apiClient.invokeAPI({{localVariablePrefix}}path, "{{httpMethod}}", {{localVariablePrefix}}queryParams, {{localVariablePrefix}}postBody, {{localVariablePrefix}}headerParams, {{localVariablePrefix}}formParams, {{localVariablePrefix}}accept, {{localVariablePrefix}}contentType, {{localVariablePrefix}}authNames, null);
+    {{localVariablePrefix}}apiClient.invokeAPI({{localVariablePrefix}}path, "{{httpMethod}}", {{localVariablePrefix}}queryParams, {{localVariablePrefix}}postBody, {{localVariablePrefix}}postBinaryBody, {{localVariablePrefix}}headerParams, {{localVariablePrefix}}formParams, {{localVariablePrefix}}accept, {{localVariablePrefix}}contentType, {{localVariablePrefix}}authNames, null);
     {{/returnType}}
+    {{/isBinary}}
+
+    {{/isDefault}}
+    {{/responses}}
```

Comment:
@boazsapir the above changes seem to make integration tests fail for the Petstore sample:

```shell
mvn clean package
./bin/java-petstore.sh
cd samples/client/petstore/java/default
mvn clean test
```

```
Results :

Failed tests:   testFindPetsByStatus(io.swagger.petstore.test.PetApiTest)
  testFindPetsByTags(io.swagger.petstore.test.PetApiTest)

Tests in error:
  testUpdatePetWithForm(io.swagger.petstore.test.PetApiTest): {"code":1,"type":"error","message":"Pet not found"}
  testDeletePet(io.swagger.petstore.test.PetApiTest): {"code":1,"type":"error","message":"Pet not found"}
  testCreateAndGetPet(io.swagger.petstore.test.PetApiTest): {"code":1,"type":"error","message":"Pet not found"}
  testUpdatePet(io.swagger.petstore.test.PetApiTest): {"code":1,"type":"error","message":"Pet not found"}

Tests run: 33, Failures: 2, Errors: 4, Skipped: 0
```

In the `addPet` method of `PetApi`, the `apiClient.invokeAPI` statement is missing, so no HTTP request is sent at all.